### ISSUE TITLE
Select: Extend select with md-select-header directive

### ISSUE THREAD

**Issue by benjamincharity** [GetTerminus/terminus-ui#648]
This will allow an optional  UI change that exposes a fixed search box at the top of the select allowing users to filter large collections.

This will be very close to the autocomplete so we will need usage docs. (use autocomplete when filtering happens on the server and filtered select when filtering happens on the client?)

Might be waiting on a merge https://github.com/angular/material2/pull/7835

- https://stackblitz.com/edit/mat-select-search?file=app%2Fmat-select-search%2Fmat-select-search.component.ts
- https://github.com/OasisDigital/angular-material-search-select/blob/master/src/app/search-select/base.ts
- https://github.com/bithost-gmbh/ngx-mat-select-search
- https://github.com/angular/material2/issues/5697#issuecomment-364902502

**Comment by macjohnny** [comment]
@benjamincharity The implementation here https://github.com/angular/material2/issues/5697#issuecomment-364902502 might be interesting for you.

**Comment by benjamincharity** [comment]
Currently on hold since `ts-autocomplete` will serve all current needs.

**Comment by benjamincharity** [comment]
closing in favor of #847
